### ISSUE TITLE
Search for tags only at the start and end of ticket's text

### ISSUE THREAD

**Issue by DnRuban** [DnRuban/Taskhelper#39]

**Comment by Lebed02** [comment]
If you create a ticket with the text "test #bb test", the ticket will be created closed. Because #bb is in the middle of the ticket it is duplicated at the beginning, and since the user is already selected, the ticket is not assigned to the default user.

![Image](https://github.com/DnRuban/Taskhelper/assets/103741333/1017c54a-941f-4905-b76e-22b89da062df)

**Comment by Lebed02** [comment]
The bot should only search for tags at the beginning of the text, not in the middle or at the end. The same applies to the following action (text after the double colon "::")

Tags that are in the middle or at the end should be recognized, but not moved, and should be duplicated once at the beginning. If it's a user tag in the middle or at the end, it should be treated as a CC, not a bug assignment to that user. For tags that are at the beginning, it should move them to the beginning and not repeat them twice.

**Comment by Lebed02** [comment]
1. If you create a ticket with the text "test #bb test", the ticket will be created closed. The issue is reproduced

2. If you click on the CC and do not change anything, immediately close the CC, the ticket is updated everywhere and comes again in the channels. The content of the ticket remained the same, but the ticket was resent in all channels where it was after clicking on CC

**Comment by Lebed02** [comment]
1. If there is some text after the last tag in the ticket body and if you manually edit the ticket body by adding a space before the very first tag and save, then spaces will be created after the last tag

2. Also expecting a fix for pending tags:

If you create a ticket and postpone it, then add #з at the beginning of the ticket the ticket will continue to be postponed, the #з tag will be duplicated. After the ticket is no longer pending the #з tag with the date will remain in the body.

If you then open CC and close CC, #з with date will be moved to the beginning of the ticket, the ticket will not be pending (if the date is past)

And if to such a ticket again add to the beginning #з in the body of the ticket will be two tags #з and the ticket will not be pending

If you close and open the ticket it will return to normal state

**Comment by Lebed02** [comment]
1. If you add tag #з before the ticket text or after the last tag, if there is no text in the ticket, problems occur. Ticket is not forwarded, buttons become inactive and console errors appear. The error appears indefinitely until you remove #з manually
![Image](https://github.com/DnRuban/Taskhelper/assets/103741333/442fd90d-287a-4f81-9b3e-3b2f3310240d)

2. If a ticket is pending and #з is added before the priority, the ticket text will be deleted

**Comment by DnRuban** [comment]
Fixed the problem with incorrect scheduled tag parsing at the end of ticket's tags.

**Comment by DnRuban** [comment]
> Rejected. Not Approved. Please fix issue:
> 
> Example ticket: #з 2023-10-06 23:00 #ak #cc #bb #п1 тест #cc #bb
> 
> If you put the tag #з before the text "тест", the tags after the text will be removed

This is not a bug, if you put #з before "тест" then "тест" will be treated as a part of scheduled message, and because "тест" is not a correct date it is moved to the end of ticket tags, so the ticket will look like this: "#з 2023-10-06 23:00 #ak #cc #bb #п1 #з #cc #bb тест"
After that duplicated tags such as #з, #cc and #bb will be removed and result ticket will be: "#з 2023-10-06 23:00 #ak #cc #bb #п1 тест"
So tags #cc and #bb removed because they were considered as duplicates of already existing tags.

**Comment by Lebed02** [comment]
Rejected. Not Approved. Please fix issues:

1. For some reason a ticket came into the channel on May 3 that had no interactions, but in the channel's recent actions this ticket has a to_delete message, and this message refers to May 1

2. If there is a ticket "#o #ak #bb #п1 text #п1 1", where there is a tag #p1 at the beginning and at the end, if you change the user and his tag using the button, in the discussion chat the user and his priority will be changed, but the priority will not be changed. If you change a user's priority using the button, the changes will be shown in the discussions, but nothing will actually happen. Maybe this point should be handled differently.

**Comment by Andrej-Adamenko** [comment]
> 1. If you create a ticket with the text "test #bb test", the ticket will be created closed. The issue is reproduced

This should be in a separate issue...

**Comment by Andrej-Adamenko** [comment]
> 1. If you add tag #з before the ticket text or after the last tag, if there is no text in the ticket, problems occur. Ticket is not forwarded, buttons become inactive and console errors appear. The error appears indefinitely until you remove #з manually
>    ![Image](https://private-user-images.githubusercontent.com/103741333/272843180-442fd90d-287a-4f81-9b3e-3b2f3310240d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTUxODQ3MDEsIm5iZiI6MTcxNTE4NDQwMSwicGF0aCI6Ii8xMDM3NDEzMzMvMjcyODQzMTgwLTQ0MmZkOTBkLTI4N2EtNGY4MS05YjNlLTNiMmYzMzEwMjQwZC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNTA4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDUwOFQxNjA2NDFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05YzM1MDExMzhjNWE5ZmY2MDE2ODNjZGE2MDIwMjEyNzM0MGE2ZjM2Y2ZiZGViMTQ3OWFjNTc2OTE2ZjY1YTU4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.COx5vrJNZ9uc0RX3MuRovQN-XTftQblzH0YGSOSmPO0)

This should be in a separate issue.

**Comment by Andrej-Adamenko** [comment]
@Lebed02:
Create a separate issue for every new problem you discover.

**Comment by DnRuban** [comment]
Rebased this feature to #66 branch, updated and fixed the bugs that was found after rebase.

**Comment by Lebed02** [comment]
Rejected. Not Approved. Please fix issues:

1. If you enter a user in the following action, the user will be displayed in the tag line only after clicking on the buttons
**STR:**
_a. Create a ticket 
b. Add the following action to it, e.g: ": #cc"
c. Take a look at what the ticket looks like_
**Actual result:** _tag "#cc" was not added to the line with all tags automatically, only after clicking on the buttons it moved to the right line_ 
**Expected result:** _The tag should be automatically added in the right place without clicking on the buttons_

2. When a ticket has a pending state, it is re-sent to all channels when any button is pressed

3. If you create a ticket with the text "test #cc", after submitting this ticket will be assigned to user "#cc", but If it's a user tag in the middle or at the end, it should be treated as a CC, not a bug assignment to that user.

**Comment by DnRuban** [comment]
1. This will be fixed in #45 
2. Already fixed in #40 
3. Already fixed in #40

**Comment by Lebed02** [comment]
Approved. Issues not found. All works good.

**Comment by DnRuban** [comment]
Fixed the bug with copying priority tags.
